server/schemas/resolvers.js

```diff
@@ -58,7 +58,10 @@ const resolvers = {
       
     },
     deleteSupplier: async (parent, { _id }, context) => {
-      
+        if (context.user) {
+            return Supplier.findOneAndDelete({ _id});
```

Comment:
We forgot the await here. 

```suggestion
            return await Supplier.findOneAndDelete({ _id});
```